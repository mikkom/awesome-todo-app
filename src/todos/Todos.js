import React, { Component } from 'react';
import Todo from './Todo';
import InputBar from '../components/InputBar';
import * as api from './api';

class Todos extends Component {
  constructor(props) {
    super(props);
    this.state = {
      todos: []
    }
  }

  componentDidMount() {
    api.getTodos().then(todos => this.setState({ todos }));
  }

  addTodo = name => {
    /*
    const todo = {
      id: latestId++,
      name,
      done: false
    };
    const todos = [...this.state.todos, todo];
    this.setState({ todos });
    */
  }

  removeTodo = id => {
    /*
    const todos = this.state.todos.filter(todo => todo.id !== id);
    this.setState({ todos });
    */
  }

  toggleTodo = id => {
    /*
    const todo = this.state.todos.find(todo => todo.id === id);
    todo.done = !todo.done;
    this.setState({ todos });
    */
  }

  render() {
    return (
      <div className="form-group">
        <InputBar onSubmit={this.addTodo} />
        <ul className="list-group">
          {this.state.todos.map(todo =>
            <Todo
              key={todo.id}
              todo={todo}
              onToggleTodo={this.toggleTodo}
              onRemoveTodo={this.removeTodo}
            />
          )}
        </ul>
      </div>
    );
  }
}

export default Todos;
