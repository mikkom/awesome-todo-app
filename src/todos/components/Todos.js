import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import uuid from 'node-uuid';
import Todo from './Todo';
import InputBar from '../../components/InputBar';
import * as api from '../api';
import { getTodos } from '../selectors';
import {
  startAdd,
  startToggle,
  startRemove,
  setTodos
} from '../actions';

export class Todos extends Component {
  componentDidMount() {
    api.getTodos().then(this.props.setTodos);
  }

  addTodo = name => {
    const todo = {
      id: uuid.v4(),
      name,
      done: false
    };
    this.props.startAdd(todo);
  }

  render() {
    return (
      <div className="form-group">
        <InputBar onSubmit={this.addTodo} />
        <ul className="list-group">
          {this.props.todos.map(todo =>
            <Todo
              key={todo.id}
              todo={todo}
              onToggleTodo={this.props.startToggle}
              onRemoveTodo={this.props.startRemove}
            />
          )}
        </ul>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  todos: getTodos(state)
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    startAdd,
    startToggle,
    startRemove,
    setTodos
  }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Todos);
