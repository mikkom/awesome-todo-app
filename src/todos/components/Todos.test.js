import React from 'react';
import { shallow } from 'enzyme';
import { Todos } from './Todos';

const todos = [
  {
    id: 1,
    name: 'write tests',
    done: true
  },
  {
    id: 2,
    name: 'write more tests',
    done: false
  },
  {
    id: 3,
    name: 'enjoy',
    done: false
  },
]

it('renders without crashing', () => {
  shallow(<Todos todos={todos} />);
});

it('matches snapshot', () => {
  expect(shallow(<Todos todos={todos} />)).toMatchSnapshot();
});
