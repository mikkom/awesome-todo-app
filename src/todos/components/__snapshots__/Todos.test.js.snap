exports[`test matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="form-group">
    <InputBar
        onSubmit={[Function]} />
    <ul
        className="list-group">
        <Todo
            onRemoveTodo={undefined}
            onToggleTodo={undefined}
            todo={
                Object {
                    "done": true,
                    "id": 1,
                    "name": "write tests",
                  }
            } />
        <Todo
            onRemoveTodo={undefined}
            onToggleTodo={undefined}
            todo={
                Object {
                    "done": false,
                    "id": 2,
                    "name": "write more tests",
                  }
            } />
        <Todo
            onRemoveTodo={undefined}
            onToggleTodo={undefined}
            todo={
                Object {
                    "done": false,
                    "id": 3,
                    "name": "enjoy",
                  }
            } />
    </ul>
</div>,
  "nodes": Array [
    <div
      className="form-group">
      <InputBar
            onSubmit={[Function]} />
      <ul
            className="list-group">
            <Todo
                  onRemoveTodo={undefined}
                  onToggleTodo={undefined}
                  todo={
                        Object {
                              "done": true,
                              "id": 1,
                              "name": "write tests",
                            }
                  } />
            <Todo
                  onRemoveTodo={undefined}
                  onToggleTodo={undefined}
                  todo={
                        Object {
                              "done": false,
                              "id": 2,
                              "name": "write more tests",
                            }
                  } />
            <Todo
                  onRemoveTodo={undefined}
                  onToggleTodo={undefined}
                  todo={
                        Object {
                              "done": false,
                              "id": 3,
                              "name": "enjoy",
                            }
                  } />
      </ul>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Todos
        todos={
                Array [
                        Object {
                          "done": true,
                          "id": 1,
                          "name": "write tests",
                        },
                        Object {
                          "done": false,
                          "id": 2,
                          "name": "write more tests",
                        },
                        Object {
                          "done": false,
                          "id": 3,
                          "name": "enjoy",
                        },
                      ]
        } />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Todos {
        "_reactInternalInstance": [Circular],
        "addTodo": [Function],
        "context": Object {},
        "props": Object {
          "todos": Array [
            Object {
              "done": true,
              "id": 1,
              "name": "write tests",
            },
            Object {
              "done": false,
              "id": 2,
              "name": "write more tests",
            },
            Object {
              "done": false,
              "id": 3,
              "name": "enjoy",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="form-group">
          <InputBar
                    onSubmit={[Function]} />
          <ul
                    className="list-group">
                    <Todo
                              onRemoveTodo={undefined}
                              onToggleTodo={undefined}
                              todo={
                                        Object {
                                                  "done": true,
                                                  "id": 1,
                                                  "name": "write tests",
                                                }
                              } />
                    <Todo
                              onRemoveTodo={undefined}
                              onToggleTodo={undefined}
                              todo={
                                        Object {
                                                  "done": false,
                                                  "id": 2,
                                                  "name": "write more tests",
                                                }
                              } />
                    <Todo
                              onRemoveTodo={undefined}
                              onToggleTodo={undefined}
                              todo={
                                        Object {
                                                  "done": false,
                                                  "id": 3,
                                                  "name": "enjoy",
                                                }
                              } />
          </ul>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="form-group">
          <InputBar
                    onSubmit={[Function]} />
          <ul
                    className="list-group">
                    <Todo
                              onRemoveTodo={undefined}
                              onToggleTodo={undefined}
                              todo={
                                        Object {
                                                  "done": true,
                                                  "id": 1,
                                                  "name": "write tests",
                                                }
                              } />
                    <Todo
                              onRemoveTodo={undefined}
                              onToggleTodo={undefined}
                              todo={
                                        Object {
                                                  "done": false,
                                                  "id": 2,
                                                  "name": "write more tests",
                                                }
                              } />
                    <Todo
                              onRemoveTodo={undefined}
                              onToggleTodo={undefined}
                              todo={
                                        Object {
                                                  "done": false,
                                                  "id": 3,
                                                  "name": "enjoy",
                                                }
                              } />
          </ul>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Todos
    todos={
        Array [
            Object {
              "done": true,
              "id": 1,
              "name": "write tests",
            },
            Object {
              "done": false,
              "id": 2,
              "name": "write more tests",
            },
            Object {
              "done": false,
              "id": 3,
              "name": "enjoy",
            },
          ]
    } />,
}
`;
